### PR TITLE
rescue HTTP request errors with pretty message 3444

### PR DESCRIPTION
**Pull request by thutterer**
[:bug: 982630](https://bugzilla.suse.com/show_bug.cgi?id=982630)

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
@sandrew will review your pull request :dancers: check https://trello.com/c/xwwGYWVx/3444-3-suse-connect-improve-error-messages
  @sandrew: Please review this pull request using our guidelines:
  * test for acceptance criteria / functionality
  * check if the new code is covered with tests
  * check for unintended consequences
  * encourage usage of the boyscout rule
  * make sure the code is architected in the best way
  * check that no unnecessary technical debt got introduced
  * make sure that no unnecessary FIXMEs or TODOs got added

File: lib/suse/connect/connection.rb [hunk 1]
```diff
@@ -66,6 +66,9 @@ def json_request(method, path, params = {})
           body: body,
           success: response.is_a?(Net::HTTPSuccess)
         )
+
+      rescue Zlib::Error
+        raise SUSE::Connect::NetworkError, 'Network error. Try again and report a bug if it keeps happening.'
```
Comment (kirushik) [inline lib/suse/connect/connection.rb]: Can we add a link to a new bug creation page here, or something among those lines?

Comment (thutterer) [inline lib/suse/connect/connection.rb]: I don't think so. According to [this comment](https://bugzilla.suse.com/show_bug.cgi?id=982630#c3) all what is asked for is a generic _"please try again"_ message. Other messages like this in YaST or Connect have no bugzilla link either.

Comment (sandrew) [inline lib/suse/connect/connection.rb]: 1) `Network error.` is not very useful tbh. Something like `Check your network connection` would be better imho.
2) `and` in second sentence confuses me. It invites me to report bug before I read the condition. `Or` or re-ordering words would be better imho.

Comment (thutterer) [inline lib/suse/connect/connection.rb]: `Check your network connection and try again. If it keeps failing, report a bug.` better?

Comment (sandrew) [inline lib/suse/connect/connection.rb]: Yep :+1:

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-Connect-Prophet/3/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged e2b8473af377b67db71e45e89a849068ad0c18d6 into 2f8cf10bcb040f27f476d5f4f7eb3b374b51464b )

**Pull request merged**